### PR TITLE
use new selector structure

### PR DESCRIPTION
**Pull request by cescoferraro**
This PR...

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/src/constants/Filter.constants.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import {Filter} from '../types/Common.types';
+
+export const filterList: Filter[] = [
```
Comment (xoscar) [inline web/src/constants/Filter.constants.ts:3]: What is this for?

File: web/src/services/Selector.service.ts [hunk 2]
```diff
@@ -43,50 +32,34 @@ const getPseudoSelectorString = (pseudoSelector?: TPseudoSelector): string => {
...
-
-    const pseudoSelector = selectorString.substring(selectorString.indexOf(']:') + 1);
-
-    if (pseudoSelector.includes('nth_child')) {
-      const [number] = (pseudoSelector.match(nthChildNumberRegex) || []).reverse();
-
-      return {selector: PseudoSelector.NTH, number: Number(number)};
-    }
-
+  getPseudoSelector(structure: Structure[]): TPseudoSelector | undefined {
```
Comment (xoscar) [inline web/src/services/Selector.service.ts:59]: Are we covering the possibility of having multiple pseudo selectors?

**Review by xoscar — CHANGES_REQUESTED** [review]
Hey @cescoferraro gj man, I left some questions and concerns. Let me know what you think

File: web/src/services/Selector.service.ts (hunk 2 above)
Comment (cescoferraro) [reply web/src/services/Selector.service.ts:59]: i think we do not use that yet. this functionality would have to be added when implementing the ancestor functionality

File: web/src/constants/Filter.constants.ts (hunk 1 above)
Comment (xoscar) [reply web/src/constants/Filter.constants.ts]: I think we should move it away from the actual executable code and have a separated section for test utils, etc.

File: web/src/services/Selector.service.ts (hunk 2 above)
Comment (xoscar) [reply web/src/services/Selector.service.ts]: Ok, let's just not lose track of that

Comment (xoscar) [reply web/src/services/Selector.service.ts]: I think either way we have to consider the multiple pseudo selectors as I think that's a separated issue from the ancestors

File: web/src/models/AssertionResults.model.ts [hunk 3]
```diff
@@ -7,13 +7,13 @@ import AssertionResult from './AssertionResult.model';
...
-    resultList: results.map(({selector: {query: selector = ''} = {}, results: resultList = []}) => ({
+    resultList: results.map(({selector: {query: selector = '', structure = []} = {}, results: resultList = []}) => ({
       id: uniqueId(),
       selector,
       spanIds: AssertionService.getSpanIds(resultList),
       originalSelector: selector,
-      pseudoSelector: SelectorService.getPseudoSelector(selector),
-      selectorList: SelectorService.getSpanSelectorList(selector),
+      pseudoSelector: SelectorService.getPseudoSelector(structure),
+      selectorList: SelectorService.getSpanSelectorList(structure),
```
[the remark this answers is not in the document]
Comment (cescoferraro) [reply web/src/models/AssertionResults.model.ts]: done

File: web/src/services/Selector.service.ts (hunk 2 above)
Comment (cescoferraro) [reply web/src/services/Selector.service.ts]: i dont think there is multiple pseudo selectors. there is a single pseudo selector for each selector/structure

**Review by xoscar — APPROVED** [review]

File: web/src/models/AssertionResults.model.ts [hunk 4]
```diff
@@ -7,13 +7,13 @@ import AssertionResult from './AssertionResult.model';
...
   return {
     allPassed,
-    resultList: results.map(({selector: {query: selector = ''} = {}, results: resultList = []}) => ({
+    resultList: results.map(({selector: {query: selector = '', structure = []} = {}, results: resultList = []}) => ({
       id: uniqueId(),
       selector,
       spanIds: AssertionService.getSpanIds(resultList),
       originalSelector: selector,
-      pseudoSelector: SelectorService.getPseudoSelector(selector),
-      selectorList: SelectorService.getSpanSelectorList(selector),
```
[the remark this answers is not in the document]
Comment (cescoferraro) [reply web/src/models/AssertionResults.model.ts:16]: Oscar asked me to keep it

**Pull request merged**